common/src/main/java/com/microsoft/identity/common/internal/controllers/ExceptionAdapter.java

```diff
@@ -188,24 +198,143 @@ public static ServiceException getExceptionFromTokenErrorResponse(@NonNull final
         }
 
         outErr.setOauthSubErrorCode(errorResponse.getSubError());
+        setHttpResponseUsingTokenErrorResponse(outErr, errorResponse);
+        return outErr;
+    }
 
-        try {
-            outErr.setHttpResponse(
-                    synthesizeHttpResponse(
-                            errorResponse.getStatusCode(),
-                            errorResponse.getResponseHeadersJson(),
-                            errorResponse.getResponseBody()
-                    )
+
+    public static ServiceException getExceptionFromTokenErrorResponse(@Nullable final CommandParameters commandParameters,
```

Comment:
nit: it might be cleaner to rename this to "getIntunePolicyRequiredException", and in the original getExceptionFromTokenErrorResponse
- invoke this function
- if it doesn't return anything, proceed with the original logic.